best-practices/native/react-native/common-components/buttons/button.tsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import { StyleSheet, TouchableOpacity, Text, Image } from "react-native";
+import PropTypes from "prop-types";
+
+interface ButtonProps {
+  onPress: Function;
+  text: string;
+  icon: any;
+  height: number;
+  width: number;
+  backgroundColor: string;
+}
+
+type InternalOptions = {
```

Comment:
Could we refine this type name a little?  These options seem to pertain only to the button component, and to make this clear without inspecting the rest of the code, we could prefix it like `ButtonInternalOptions` or `ButtonOptions`.

Reply:
YES! Thanks for coming up with that name. It seems like it would have been very easy, but alas, my brain was fighting me on that one!